tests/robot-tests/tests/libs/common.robot

```diff
@@ -28,13 +28,21 @@ do this on failure
     END
 
 custom testid locator strategy
-    [Arguments]    ${browser}    ${test_id}    ${tag}    ${constraints}
-    ${element}=    Execute Javascript
-    ...    let xPathResult = document.evaluate('//*[@data-testid="${test_id}"]', document); if(xPathResult) return xPathResult.iterateNext(); return [];
-    [Return]    ${element}
+    [Arguments]  ${browser}  ${test_id}   ${tag}  ${constraints}
+    ${element}=  user gets testid element  ${test_id}
+    [Return]  ${element}
+
+custom label locator strategy
+    [Arguments]  ${browser}  ${label}  ${tag}  ${constraints}
+    user waits until element is visible  xpath://label[text()="${label}"]
+    ${label_el}=  get web element  xpath://label[text()="${label}"]
+    ${input_id}=  get element attribute  ${label_el}  for
+    ${element}=   get web element  id:${input_id}
+    [Return]  ${element}
 
 set custom locator strategies
-    add location strategy    testid    custom testid locator strategy
+    add location strategy   testid    custom testid locator strategy
+    add location strategy   label     custom label locator strategy
```

Comment:
Just worth checking with @luke-h1 - when he put in the testid strategy, he reckoned there were some situations where it didn't work - might be worth knowing about those

Reply:
Checked with him before I did this and didn't seem like there was any concrete evidence for it not working.

Having ran all the tests with it, I haven't seen any test failures so I'd say that's good enough :smile:

Reply:
Yup good to have it in, ta!